### PR TITLE
Feat: handle tarball programs

### PR DESCRIPTION
**Pull request by echometerain**
# Purpose
A lot of programs ship as tarballs

# Approach

Unzip the tar like the .zip

### CODE REVIEW

File: misc/scripts/install-local.sh [hunk 1]
```diff
@@ -358,6 +358,19 @@ case "$url" in
...
+		hashcheck "${url##*/}"
+		# unzip file
+		sudo tar -xf "${url##*/}" 1>&1 2>/dev/null
+		# cd into it
+		cd ./*/
+		# export srcdir
+		export srcdir="/tmp/pacstall/$PWD"
+		# Make the directory available for users
+		sudo chown -R "$LOGNAME":"$LOGNAME" . 2>/dev/null
+	;;
```
Comment (D-Brox) [inline misc/scripts/install-local.sh:373]: Technically this is already done in `*)`, but I prefer it this way. Remove the `tar` and the `cd` from `*)` and we could merge.

**Review by D-Brox — CHANGES_REQUESTED** [review]

File: misc/scripts/install-local.sh [hunk 2]
```diff
@@ -393,6 +393,19 @@ case "$url" in
 		# Make the directory available for users
 		sudo chown -R "$LOGNAME":"$LOGNAME" . 2>/dev/null
 	;;
+	*.tar*)
+		download "$url"
+		# hash the file
+		hashcheck "${url##*/}"
```
Comment (wizard-28) [inline misc/scripts/install-local.sh:399]:
```suggestion	
		if ! hashcheck "${url##*/}"; then
			return 1
		fi
```

File: misc/scripts/install-local.sh [hunk 3]
```diff
@@ -425,12 +438,7 @@ case "$url" in
 		download "$url"
 		# I think you get it by now
 		hashcheck "${url##*/}"
```
Comment (wizard-28) [inline misc/scripts/install-local.sh:440]:
```suggestion	
		if ! hashcheck "${url##*/}"; then
			return 1
		fi
```

File: misc/scripts/install-local.sh (hunk 1 above)
Comment (wizard-28) [reply misc/scripts/install-local.sh:408]: Why are we adding a separate case for `.tar`? In this method in future we may have to add different cases for `.gz`, `.xz` and other types of `tar`s (which were currently automatically handled by `*)`)

**Review by wizard-28 — CHANGES_REQUESTED** [review]

**Comment by NarutoXY** [comment]
Well @wizard-28 what are plans for this pr?

**Comment by wizard-28** [comment]
I don't really know... Like I said. We are already unzipping tars...

**Comment by NarutoXY** [comment]
Yea this is like a redundant pr. We already are unzipping tars. Closing for now

**Pull request closed without merging**